frontend/rust-lib/dart-ffi/src/lib.rs

```diff
@@ -85,26 +130,52 @@ pub extern "C" fn init_sdk(_port: i64, data: *mut c_char) -> i64 {
     DEFAULT_NAME.to_string(),
   );
 
-  // Ensure that the database is closed before initialization. Also, verify that the init_sdk function can be called
-  // multiple times (is reentrant). Currently, only the database resource is exclusive.
-  if let Some(core) = &*APPFLOWY_CORE.0.lock().unwrap() {
+  if let Some(core) = &*DART_APPFLOWY_CORE.core.write().unwrap() {
     core.close_db();
   }
 
-  let runtime = Rc::new(AFPluginRuntime::new().unwrap());
-  let cloned_runtime = runtime.clone();
-
   let log_stream = LOG_STREAM_ISOLATE
-    .lock()
+    .write()
     .unwrap()
     .take()
     .map(|isolate| Arc::new(LogStreamSenderImpl { isolate }) as Arc<dyn StreamLogSender>);
-
-  // let isolate = allo_isolate::Isolate::new(port);
-  *APPFLOWY_CORE.0.lock().unwrap() = runtime.block_on(async move {
-    Some(AppFlowyCore::new(config, cloned_runtime, log_stream).await)
-    // isolate.post("".to_string());
+  let (sender, mut task_rx) = mpsc::unbounded_channel::<Task>();
+  let handle = std::thread::spawn(move || {
+    let local_set = LocalSet::new();
+    let runtime = Builder::new_current_thread().enable_all().build().unwrap();
+    runtime.block_on(async {
```

Comment:
With that I think we still execute everything in a single thread, but at least this time this thread is not UI one.

I have few ideas how we could to more parallel work here, but this can be separate PR.

Reply:
Nice. FYI. I want to fix dispatch ASAP and then improve it later.